addons/purchase_stock/models/purchase_order_line.py

```diff
@@ -152,13 +161,19 @@ def unlink(self):
     # Business methods
     # --------------------------------------------------
 
-    def _update_move_date_deadline(self, new_date):
-        """ Updates corresponding move picking line deadline dates that are not yet completed. """
+    def _update_move_dates(self, date_planned=False, date_promised=False):
```

Comment:
If we want to be really clean with the code, you should add in a:
```
if not date_planned and not date_promised
    return False
```
So that you skip all the filtering in case someone calls this method without passing either value.